Bulldozer.BinaryFile/Maps/GroupImage.cs

```diff
@@ -0,0 +1,246 @@
+                        BinaryFileTypeId = groupImageType.Id,
+                        FileName = file.Name,
+                        Description = string.Format("Imported as {0}", file.Name),
+                        CreatedDateTime = file.LastWriteTime.DateTime,
+                        ModifiedDateTime = file.LastWriteTime.DateTime,
+                        CreatedByPersonAliasId = ImportPersonAliasId
+                    };
+
+                    rockFile.SetStorageEntityTypeId( groupImageType.StorageEntityTypeId );
+                    rockFile.StorageEntitySettings = emptyJsonObject;
+
+                    if ( groupImageType.AttributeValues.Any() )
+                    {
+                        rockFile.StorageEntitySettings = groupImageType.AttributeValues
+                            .ToDictionary( a => a.Key, v => v.Value.Value ).ToJson();
+                    }
+
+                    // use base stream instead of file stream to keep the byte[]
+                    // NOTE: if byte[] converts to a string it will corrupt the stream
+                    using ( var fileContent = new StreamReader( file.Open() ) )
+                    {
+                        rockFile.ContentStream = new MemoryStream( fileContent.BaseStream.ReadBytesToEnd() );
+                    }
+
+                    newFileList.Add( new GroupDocumentKeys()
+                    {
+                        GroupId = group.Id,
+                        AttributeId = groupImageAttribute.Id,
+                        File = rockFile
+                    });
+
+                    completedItems++;
+                    if ( completedItems % percentage < 1 )
+                    {
+                        var percentComplete = completedItems / percentage;
+                        ReportProgress( percentComplete, string.Format( "{0:N0} group image files imported ({1}% complete).", completedItems, percentComplete ) );
+                    }
+
+                    if ( completedItems % ReportingNumber < 1 )
+                    {
+                        SaveFiles(newFileList);
+
+                        // Reset list
+                        newFileList.Clear();
+                        ReportPartialProgress();
+                    }
+                }
+            }
+
+            if ( newFileList.Any() )
+            {
+                SaveFiles(newFileList);
+            }
+
+            ReportProgress( 100, string.Format( "Finished group images import: {0:N0} group images imported.", completedItems ) );
+            return completedItems;
+        }
+
+        /// <summary>
+        /// Saves the files.
+        /// </summary>
+        /// <param name="newFileList">The new file list.</param>
+        private static void SaveFiles(List<GroupDocumentKeys> newFileList)
+        {
+            var rockContext = new RockContext();
+            rockContext.WrapTransaction( () =>
+            {
+                rockContext.BinaryFiles.AddRange(newFileList.Select(f => f.File));
```

Comment:
add check to condition

Reply:
@guymaness2 did this get taken care of?

Reply:
need to copy null check from line 199 to this one as well.